### PR TITLE
Move system test runner from bash to Python

### PR DESCRIPTION
**Pull request by dhermes**
Also re-tooling the `run_module_tests` method to be more useful from other directories (so that we could call it from the `scripts/` subdirectory).

### CODE REVIEW

**Comment by dhermes** [comment]
Towards #1123

File: system_tests/bigquery.py [hunk 1]
```diff
@@ -23,12 +23,15 @@
 
 
 _helpers.PROJECT = TESTS_PROJECT
-CLIENT = bigquery.Client()
```
Comment (tseaver) [inline system_tests/bigquery.py]: How are these changes related to the intent of the PR?

File: system_tests/datastore.py [hunk 2]
```diff
@@ -25,8 +25,18 @@
...
+
+    This is a mutable stand-in to allow test set-up to modify
+    global state.
+    """
+    CLIENT = None
+
+
+def setUpModule():
+    client.DATASET = TESTS_DATASET
+    Config.CLIENT = datastore.Client()
```
Comment (tseaver) [inline system_tests/datastore.py]: Why `Config.CLIENT` here and `self.client` in `bigquery.py`?  Shouldn't we follow a consistent pattern?

File: system_tests/run_system_test.py [hunk 3]
```diff
@@ -18,14 +18,25 @@
...
 # repository root is the current directory.
+from system_tests import bigquery
+from system_tests import datastore
+from system_tests import pubsub
+from system_tests import storage
 from system_tests import system_test_utils
 
+
 REQUIREMENTS = {
     'datastore': ['dataset_id', 'credentials'],
```
Comment (tseaver) [inline system_tests/run_system_test.py]: This bit is going to change with `v1b3`, no?

File: system_tests/bigquery.py (hunk 1 above)
Comment (dhermes) [inline system_tests/bigquery.py]: See the third commit message.

> Updating system tests so configuration is done at runtime.
>
>In a set-up where the tests are not intended to be run (e.g. a PR build for Travis), just importing these modules caused errors due to lack of implicit variables being set.
>
> So, all configuration done at import time was moved into `setUpClass` if there was only one `TestCase` or moved to `setUpModule` with some mutable global if there were multiple `TestCase`s.

File: system_tests/datastore.py (hunk 2 above)
Comment (dhermes) [inline system_tests/datastore.py]: Sure thing. I didn't want to use `Config` but also didn't want to use the `global` keyword. What do you think would be best?

File: system_tests/run_system_test.py (hunk 3 above)
Comment (dhermes) [inline system_tests/run_system_test.py]: Yup. It's on the checklist in #1288.

File: system_tests/bigquery.py (hunk 1 above)
Comment (tseaver) [inline system_tests/bigquery.py]: What was causing these quasi-modules to be imported?

File: system_tests/datastore.py (hunk 2 above)
Comment (tseaver) [inline system_tests/datastore.py]: The `Config` class seems to be the more general pattern (simpler than refactoring the muli-testcase scripts to use a single base class).

File: system_tests/bigquery.py (hunk 1 above)
Comment (dhermes) [inline system_tests/bigquery.py]: On a Travis build for a PR ([e.g.][1]) we used to run `run_system_tests.sh` and it would bail out since we only run system tests for merged commits to master (not PRs).

Now this check happens in Python.

This also means the Python script that does the environment checking imports `system_tests.run_system_test`. That import in turn imports `system_tests.datastore`, `system_tests.storage`, etc.

But a (security) feature of Travis means that PRs don't have the configuration needed to just call `datastore.Client()` and get all the implicit information.

[1]: https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/100939831

File: system_tests/datastore.py (hunk 2 above)
Comment (dhermes) [inline system_tests/datastore.py]: Implementing now. Will ping once done.

**Comment by dhermes** [comment]
@tseaver PTAL I added another commit to have all 4 system tests use `Config`.

**Comment by tseaver** [comment]
LGTM
